macros/union_data.sql

```diff
@@ -7,82 +7,102 @@
     default_database, 
     default_schema, 
     default_variable,
+    column_macro,
     union_schema_variable,
     union_database_variable
     ) }}
 
-{%- endmacro %}
+{%- endmacro -%}
 
-{% macro default__union_data(
+{%- macro default__union_data(
     table_identifier, 
     database_variable, 
     schema_variable, 
     default_database, 
     default_schema, 
     default_variable,
+    column_macro,
     union_schema_variable,
     union_database_variable
-    ) %}
+    ) -%}
 
-{% if var(union_schema_variable, none) %}
+{%- if var(union_schema_variable, none) -%}
 
-    {% set relations = [] %}
+    {%- set relations = [] -%}
     
-    {% if var(union_schema_variable) is string %}
-    {% set trimmed = var(union_schema_variable)|trim('[')|trim(']') %}
-    {% set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") %}
-    {% else %}
-    {% set schemas = var(union_schema_variable) %}
-    {% endif %}
-
-    {% for schema in var(union_schema_variable) %}
-    {% set relation=adapter.get_relation(
+    {%- if var(union_schema_variable) is string -%}
+    {%- set trimmed = var(union_schema_variable)|trim('[')|trim(']') -%}
+    {%- set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") -%}
+    {%- else -%}
+    {%- set schemas = var(union_schema_variable) -%}
+    {%- endif -%}
+
+    {%- for schema in var(union_schema_variable) -%}
+    {%- set relation=adapter.get_relation(
         database=source(schema, table_identifier).database if var('has_defined_sources', false) else var(database_variable, default_database),
         schema=source(schema, table_identifier).schema if var('has_defined_sources', false) else schema,
         identifier=source(schema, table_identifier).identifier if var('has_defined_sources', false) else table_identifier
     ) -%}
     
-    {% set relation_exists=relation is not none %}
+    {%- set relation_exists=relation is not none -%}
 
-    {% if relation_exists %}
+    {%- if relation_exists -%}
+        {%- do relations.append(relation) -%}
+    {%- endif -%}
 
-    {% do relations.append(relation) %}
+    {%- endfor -%}
     
-    {% endif %}
-
-    {% endfor %}
-
-    {{ dbt_utils.union_relations(relations) }}
-
-{% elif var(union_database_variable, none) %}
+    {%- if relations != [] -%}
+        {{ dbt_utils.union_relations(relations) }}
+    {%- else -%}
+    select 
+        {% if column_macro != [{}] %}{{ fivetran_utils.fill_staging_columns(source_columns=[], staging_columns=column_macro) }},{% endif -%} 
```

Comment:
As I stated in the [shopify_source review](https://github.com/fivetran/dbt_shopify_source/pull/59), I think this is possibly duplicative. I think we can remove the column_macro since we can gain comfort that the staging model will ensure success and parity of fields.

Additionally, this results in less maintenance across the packages!

Reply:
removed